Feature/correlating bpmn messages (#289)
valtimo-platform/valtimo-documentation: using-valtimo/process/correlating-messages.md

# Correlating messages in BPMN

When modeling a process, correlating a message (e.g. to start another process) does not do anything by default. Valtimo
offers several methods to facilitate message correlation. These can be separated into starting a process

> (Message Start Events) and message catch events (Message Boundary Events and Message Intermediate Catch Events).

Reviewer: ```suggestion
(Message Start Events) and receiving messages in a running process (Message Boundary Events and Message Intermediate Catch Events).
```

## How to use

Valtimo provides several methods that can be used inside a BPMN, accessible through the `correlationService` bean. Which
method to use depends on the use case. For more information on the separate use cases see the two sections below.

These methods can be used in expressions applied to message throw events like this:

![intermediate-throw-example](img/intermediate-throw-event.png)

> The first argument always corresponds to the message the is supposed to send. In this example, there should also be a

Reviewer: ```suggestion
The first argument is the key of the message that should be send. In this example, there should also be a
```

Reviewer 2: ```suggestion
The first argument is the key of the message that should be sent. In this example, there should also be a
```
